dss_metrics/minio_rest_collector.py

```diff
@@ -64,24 +64,28 @@ def poll_statistics(self, metrics_data_buffer, exit_flag):
                 if not endpts or len(endpts) == 0:
```

Comment:
Since the endpts is same across each run, it will be better to move it to the top so that we are not running items() call again and again on the same data thereby reducing some cpu cycles
Also add additional checks in case of failure to get the endpoint_map or endpoints